packages/vuetify/src/services/theme/index.ts

```diff
@@ -228,24 +229,11 @@ export class Theme extends Service {
     // Only watch for reactivity on client side
     if (typeof document === 'undefined') return
 
-    // If we get here somehow, ensure
-    // existing instance is removed
-    if (this.vueInstance) this.vueInstance.$destroy()
-
-    // Use Vue instance to track reactivity
-    // TODO: Update to use RFC if merged
-    // https://github.com/vuejs/rfcs/blob/advanced-reactivity-api/active-rfcs/0000-advanced-reactivity-api.md
-    this.vueInstance = new Vue({
-      data: { themes: this.themes },
-
-      watch: {
-        themes: {
-          immediate: true,
-          deep: true,
-          handler: () => this.applyTheme(),
-        },
-      },
-    })
+    this.applyTheme()
```

Comment:
Use `immediate: true` instead?
Also with reactivity api this service could be refactored to use computeds or remove `this.applyTheme()` calls, but probably not in scope of this PR (if it is let me know, i'll try to play with it)

Reply:
It is out of scope of this PR. I do have a theme-service-refactor RFC that I've been working on though: https://www.notion.so/vuetify/0017-theme-service-refactor-894e0d7594b54ba49609c370b8e375bd

Reply:
watchers are immediate true by default so line 232 is pointless.